app/controllers/users_controller.rb

```diff
@@ -1,31 +1,78 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by id: params[:id]
-    return if @user
+  before_action :logged_in_user, except: [:new, :show, :create]
+  before_action :find_user, only: [:edit, :update, :show]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
 
-    flash[:warning] = t "shared.user_not_found"
-    redirect_to root_path
+  def index
+    @users = User.page(params[:page]).per Settings.pagination
   end
 
   def new
     @user = User.new
   end
 
+  def show; end
+
+  def edit; end
+
   def create
     @user = User.new user_params
     if @user.save
       log_in @user
-      flash[:success] = t "shared.success"
+      flash[:success] = t "shared.signup_success"
       redirect_to @user
     else
-      flash[:danger] = t "shared.danger"
+      flash[:danger] = t "shared.signup_danger"
       render :new
     end
   end
 
+  def update
+    if @user.update user_params
+      flash[:success] = t "edit.edit_success"
+      log_in @user
+      redirect_to @user
+    else
+      flash[:danger] = t "edit.edit_danger"
+      render :edit
+    end
+  end
+
+  def destroy
+    if User.find_by(id: params[:id])&.destroy
```

Comment:
```suggestion
    if @user.destroy
```
find_user sao k dùng e ?

Reply:
Dùng find_user thì bị lỗi ạ :((